File: camera.py
from pyray import *
import math

class Camera:
    zoom = 1
    camera = Camera2D(Vector2(0, 0), Vector2(0, 0), 0, zoom)

    def restart():
        Camera.camera = Camera2D(Vector2(0, 0), Vector2(0, 0), 0, Camera.zoom)

    def update(window):
        wheel = get_mouse_wheel_move()
        #mouseWorldPos = get_screen_to_world_2d(get_mouse_position(), Camera.camera)
        Camera.camera.offset = vector2_scale(window, 0.5)
        scale = 0.2 * wheel
        zoom = math.exp(math.log(Camera.camera.zoom) + scale)
        Camera.camera.zoom = clamp(zoom, 0.125, 64.0)

File: camera.go
package main

import (
	. "github.com/gen2brain/raylib-go/raylib"
	"math"
)

var (
	MainCamera Camera2D
)

func CameraInit() {
	MainCamera.Zoom = 2
}

func CameraRestart() {
	MainCamera.Zoom = 2
}

func CameraUpdate() {
	wheel := GetMouseWheelMove()
	MainCamera.Offset = Vector2Scale(WindowSize, 0.5)
	scale := 0.2 * wheel
	zoom := f32(math.Exp(math.Log(f64(MainCamera.Zoom)) + f64(scale)))
	MainCamera.Zoom = Clamp(zoom, 0.125, 64)
}

func CameraRect(offset f32) Rectangle {
	r := Rectangle{}
	r.Width = WindowSize.X / MainCamera.Zoom
	r.Height = WindowSize.Y / MainCamera.Zoom
	r.X = MainCamera.Target.X - r.Width*0.5 + offset
	r.Y = MainCamera.Target.Y - r.Height*0.5 + offset
	r.Width -= 2 * offset
	r.Height -= 2 * offset
	return r
}
